libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -2557,10 +2557,25 @@ string IRGeneratorForStatements::binaryOperation(
 		!TokenTraits::isShiftOp(_operator),
 		"Have to use specific shift operation function for shifts."
 	);
-	if (IntegerType const* type = dynamic_cast<IntegerType const*>(&_type))
+	string fun;
+	if (TokenTraits::isBitOp(_operator))
 	{
-		string fun;
-		// TODO: Implement all operations for signed and unsigned types.
+		solAssert(
+			_type.category() == Type::Category::Integer ||
+			_type.category() == Type::Category::FixedBytes,
+		"");
```

Comment:
Is indentation correct here?

Reply:
Strictly, you could ask for
```
solAssert(
  _type.cat... ||
  _type.cat... FixedBytes
, "");
```
But I think moving the `,` to the previous line is better.